Exercícios/Para casa/entregas-mariana-seidel/mariana-seidel/SavingAccount.js

```diff
@@ -0,0 +1,59 @@
+const { BankAccount } = require("./BankAccount");
+
+class SavingAccount extends BankAccount {
+    incomeRate;
+    incomeDay;
+    #qtdWithdrawal = 0;
+    MAX_OF_WITHDRAWAL = 2;
+    #withdrawalTax = 0.03
+
+    constructor(client, bank, accountNumber, agencyNumber, incomeRate, incomeDay){
+        super(client, bank, accountNumber, agencyNumber)
+        this.incomeRate  = incomeRate
+        this.incomeDay = incomeDay
+    }
+
+    get qtdWithdrawal() {
+        return this.#qtdWithdrawal
+    }
+
+    get withdrawalTax() {
+        return this.#withdrawalTax
+    }
+
+    set withdrawalTax(newWithdrawalTax) {
+        this.#withdrawalTax = newWithdrawalTax
+    }
+
+    cashWithdrawal(amount){
+        console.log(`Você já realizou ${this.#qtdWithdrawal} retiradas. Você tem direito a ${this.MAX_OF_WITHDRAWAL} retiradas gratuitas.`)
+
+        if(this.#qtdWithdrawal < this.MAX_OF_WITHDRAWAL) {
+            if(this.balance >= amount){
+                this.balance -= amount
```

Comment:
Uma ideia que fiz no meu código foi colocar em constantes as mensagens e a incrementação que se repetem, por exemplo:

const incrementQtdWithdrawal = this.#qtdWithdrawal
aí eu chamo a constante nos lugares em que ela é necessária! A mesma coisa da pra fazer com os consoles.log

Digo isso pq daí pra diminuir a repetição de código. Agora, se tá certo o que eu fiz, não sei, mas se tiver, dá pra servir de dica ahauhaua

Reply:
Ahh verdade, adorei a dica, não tinha pensado em colocar alguns consoles em constantes!